server/server.go

```diff
@@ -0,0 +1,52 @@
+// Package server implements the web server for extension update requests
+package server
+
+import (
+	"context"
+	"fmt"
+	"github.com/brave/go-update/controller"
+	"github.com/brave/go-update/extension"
+	"github.com/go-chi/chi"
+	chiware "github.com/go-chi/chi/middleware"
+	"github.com/pressly/lg"
+	"github.com/sirupsen/logrus"
+	"log"
+	"net/http"
+	"time"
+)
+
+func setupLogger(ctx context.Context) (context.Context, *logrus.Logger) {
+	logger := logrus.New()
+	//logger.Formatter = &logrus.JSONFormatter{}
+	// Redirect output from the standard logging package "log"
+	lg.RedirectStdlogOutput(logger)
+	lg.DefaultLogger = logger
+	ctx = lg.WithLoggerContext(ctx, logger)
+	return ctx, logger
+}
+
+func setupRouter(ctx context.Context, logger *logrus.Logger) (context.Context, *chi.Mux) {
+	r := chi.NewRouter()
+	r.Use(chiware.RequestID)
+	r.Use(chiware.RealIP)
+	r.Use(chiware.Heartbeat("/"))
+	r.Use(chiware.Timeout(60 * time.Second))
```

Comment:
I might recommend something like the `RequestLogger` middleware from `bat-go` that handles panic recovery / sending to sentry (in addition to request logging). If you want to reuse that middleware you can use the tagged bat-go release via `dep ensure -add github.com/brave-intl/bat-go@0.1.0` then just import the `middleware` package

Reply:
This and things other than the XML one below are implemented and pushed out.